compiler/il/OMRNode.hpp

```diff
@@ -374,6 +375,21 @@ class OMR_EXTENSIBLE Node
       return true;
       }
 
+   static bool isNotDeprecatedUnsigned(TR::ILOpCodes opvalue)
+      {
+      static const TR::ILOpCodes DeprecatedUnsignedList[] = 
+         {
+         TR::iumul, //Multiply
+         TR::buternary, TR::iuternary, TR::luternary, TR::suternary, //Ternary
+         TR::iushl, TR::lushl, //Shift
+         TR::ishfl, TR::lshfl, TR::iushfl, TR::lushfl, TR::bshfl, TR::sshfl, TR::bushfl, TR::sushfl //Shift Logical
+         };
+
+      auto begin = &DeprecatedUnsignedList[0];
+      auto end = begin + sizeof(DeprecatedUnsignedList)/sizeof(TR::ILOpCodes);
+      return std::find(begin, end, opvalue) == end;
```

Comment:
would it be more efficient to just have a switch where we have all the cases that need to fail fall through and then a default which succeeds? Rather than scanning memory constants we would then let the compile generate the most efficient jump table? Not sure, but wanted to ask if the scan is the best perf for this since it will run a lot...

Reply:
A switch statement actually saved a lot of time by reducing the generated assembly code. I am going to use a switch statement instead of loop through an array.